src/robust.jl

```diff
@@ -44,10 +44,10 @@ function trim!(x::AbstractVector; prop::Real=0.0, count::Integer=0)
         0 <= count < n/2 || throw(ArgumentError("count must satisfy 0 ≤ count < length(x)/2."))
     end
 
-    partialsort!(x, (n-count+1):n)
-    partialsort!(x, 1:count)
-    deleteat!(x, (n-count+1):n)
-    deleteat!(x, 1:count)
+    ix = vcat(partialsortperm(x, 1:count), partialsortperm(x, (n-count+1):n))
```

Comment:
Since `partialsortperm(x, 1:count)` is going to make a copy of `x`, better do it manually, and call `partialsortperm!` on the result twice. That way you avoid an allocation, and you take advantage of the partial sorting of the first call to avoid doing some work twice.

Reply:
done